src/pages/Login/index.tsx

```diff
@@ -36,7 +36,8 @@ export default function Login() {
       .finally(() => {
         setLoading(false);
       })
-  });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```

Comment:
You can wrap handleSuccess in a useCallback to prevent infinite loading without overriding eslint, same thing for `Analytics/index.ts `

Reply:
Would replacing [] with [handleSuccess] also work?

Reply:
Yeah that's basically what I was referring to but you need to wrap handle success with usecallbsck otherwise it'll cause an infinite render loop

Reply:
Hmm, I don't think the empty [] would have caused infinite re-rendering though, but the \[handleSuccess\] prob will cause inf rendering. I added the useCallback for the Login anyway, but empty [] might be a bit clearer. For Analytics, there is no fun re-definition though, so should be fine by itself.

Reply:
Empty callback in this case is functionally the same as having a memoized handle success with no dependencies. It's generally not a good practice to override reacts exhaustive dependency warning if you can avoid it, easy to run into bugs down the line.

Reply:
Hmm, but no array and empty array are weirdly not the same though :D Ah well, added the useCallback in though, so will resolve this then.

Reply:
If you don't provide a dep array use effect will run on every render by default. Since Ur updating state within the effect that's what's causing the infinite re render.

Reply:
Why do we need a useCallback on the handleSuccess, instead of replacing only [] with [handleSuccess] and keeping the current handleSuccess arrow definition?

Reply:
Ah yeah, but eslint requires full dependency or no dep array though. The option for empty array seems to just be ignored :|

Reply:
The [handleSuccess] doesn't work, because apparently during re-render, a new handleSuccess function is generated, triggered useEffect again, I think :D